### DISCUSSION TITLE
[RESOLVED] Any way to resolve dark mode page flicker navigating between pages?

### DISCUSSION THREAD

**Discussion by gacharles23** [adonisjs/core#4379] [General]
Is there any way to avoid page flicker using Edge pages in dark mode (Tailwind) when navigating between pages? Currently I’m using Javascript in the page header to set (and maintain) the theme using local storage. On-page updates with HTMX don't cause any flicker (since only portions of the page are being replaced), but navigating from one page to another (in dark mode) leads to the dreaded FOUC on each transition. Is this a limitation of using Edge for the front-end? It would be great if there was a way to solve this as I'd love to stick with Edge.

**Comment by tomgobich** [comment]
Are you having this in production or just in development? If it's just in development, it's likely due to how Vite is plopping the styles onto the page. In development, it'll plop all the styles inside a `style` tag and use a built-out `.css` file with a `link` tag in production.

**Reply by gacharles23** [reply]
That’s an interesting thought. I haven’t deployed to a hosting service yet, so this is just development. I’m curious if there’s a solution for development though as the navigation in dark mode is quite jarring and frankly, looks unpolished. In the meantime, I will go through the build and deploy flow and report back if the issue still persists in prod. Thanks for the suggestion.

**Reply by tomgobich** [reply]
Yeah, I believe it's plopped into a `style` tag because the `app.css` is being imported within `resources/js/app.js`. If you split it out into its own entry point in Vite (or Encore if you're on 5), it looks like it'll add it in as a CSS file. Just gave it a go on the Let's Learn AdonisJS 6 codebase and I'm no longer seeing FOUC in development.

Within `app.js` comment out or remove the CSS import
```js
// import '../css/app.css'
```

Within `vite.config.js` add the CSS file as an entry point
```js
import { defineConfig } from 'vite'
import adonisjs from '@adonisjs/vite/client'

export default defineConfig({
  plugins: [
    adonisjs({
      entrypoints: ['resources/js/app.js', 'resources/css/app.css'],
      
       // ...
    }),
  ],
})
```

Within your layout or pages, add it to the `@vite` tag array
```html
<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <title> {{ title || "Let's Learn AdonisJS 6" }} </title>

  @vite(['resources/js/app.js', 'resources/css/app.css'])
</head>
<body>
  <div class="max-w-3xl mx-auto mt-6">
    @include('partials/nav')

    {{{ await $slots.main() }}}
  </div>
</body>
```

**Reply by gacharles23** [reply]
You are amazing!!! Thank you very much. Coincidentally, my deployment to AWS (Elastic Beanstalk) just finished and, sure enough, to your first comment, it works just fine in production. No FOUC.

**Reply by tomgobich** [reply]
Anytime!! Awesome, I'm happy to hear you're not having FOUC on production! 😄

**Comment by dimyatiutoyo** [comment]
I followed the instructions, but why the problem still occurred? still flickering. I use [daisyui](https://daisyui.com/) and [theme-changer](https://github.com/saadeghi/theme-change)
